### PR TITLE
pulse counter: Show how calculations are made

### PR DESCRIPTION
**Pull request by fredrike**
## Description:

The examples for pulse counting are not 100% easy to follow. This adds more information.

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by fredrike** [comment]
Would it be ok to mention a S0 connection like this for in the power_meter doc?

>
> Some meters have an available S0 port, if that is the case the photo diode can be replaced with the following connection.

```
S0 ------------- VCC
S0 ---+-- 10k -- GND
      |
      +--------- GPIO12
```

**Comment by fabaff** [comment]
> Would it be ok to mention a S0 connection like this for in the power_meter doc?

Sure, everything that helps should be included.

**Comment by fredrike** [comment]
I'm not sure why linting fails..

**Comment by fredrike** [comment]
Changed to current as the upstream changes already have been merged.

File: cookbook/power_meter.rst [hunk 1]
```diff
@@ -24,6 +24,18 @@ And... that should already be it :)
     :align: center
     :width: 80.0%
 
+.. note::
+
+    Some energy meters have an exposed S0 port (which essentially just is a switch that closes), if 
+    that is the case the photo diode can be replaced with the following connection.
```
Comment (frenck) [inline cookbook/power_meter.rst]:
```suggestion
    Some energy meters have an exposed S0 port (which essentially just is a switch that closes), if
    that is the case the photodiode can be replaced with the following connection.
```

Comment (frenck) [reply cookbook/power_meter.rst]: Additionally this kinda unrelated to the PR presented... Should this have been a separate PR?

Comment (fredrike) [reply cookbook/power_meter.rst]: Well it could, but this is more of an "update doc" pr so perhaps we could just update the name.

**Review by frenck — APPROVED** [review]
Thanks, @fredrike 👍

**Pull request merged**